### PR TITLE
perf(RuleReferences): RHICOMPL-3493 swap with RuleReferencesContainer

### PR DESCRIPTION
**Pull request by skateman**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: app/graphql/types/interfaces/rules_preload.rb [hunk 1]
```diff
@@ -68,12 +68,9 @@ def initialize_rules_context(rules, rule_results, args = {})
...
         rule_ids = rule_results.pluck(:rule_id)
-        grouped_rules_references = ::RuleReferencesRule.distinct.where(
-          rule_id: rule_ids
-        ).group_by(&:rule_id)
-        grouped_rules_references.each do |rule_id, references|
-          context[:"rule_references_#{rule_id}"] =
-            references.pluck(:rule_reference_id)
+
+        RuleReferencesContainer.where(rule_id: rule_ids).find_each do |reference|
+          context[:"rule_references_#{container.rule_id}"] = reference.rule_references
```
Comment (marleystipich2) [inline app/graphql/types/interfaces/rules_preload.rb:73]: What is `container` on this line? I don't see it being defined anywhere but maybe I'm missing it

File: test/graphql/queries/rule_query_test.rb [hunk 2]
```diff
@@ -21,10 +21,14 @@ class RuleQueryTest < ActiveSupport::TestCase
...
-      rule_references: [@rr]
-    )
+
+    @rr = rule.rule_references_container.rule_references.first
+    rule.rule_references_container.update(rule_references: [@rr])
+    # @rr = FactoryBot.create(
+    #   :rule_references_container,
+    #   rule_id: @profile.rules.first.id,
+    #   reference_count: 1
+    # ).rule_references.first
```
Comment (marleystipich2) [inline test/graphql/queries/rule_query_test.rb:31]: Did you mean to remove this?

File: app/graphql/types/interfaces/rules_preload.rb (hunk 1 above)
Comment (skateman) [reply app/graphql/types/interfaces/rules_preload.rb]: Yeah, this was a bug :see_no_evil:

File: test/graphql/queries/rule_query_test.rb (hunk 2 above)
Comment (skateman) [reply test/graphql/queries/rule_query_test.rb]: Yup

**Review by romanblanco — APPROVED** [review]

**Pull request merged**